addOns/allinonenotes/src/main/java/org/zaproxy/zap/extension/allinonenotes/ExtensionAllInOneNotes.java

```diff
@@ -0,0 +1,261 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.allinonenotes;
+
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.swing.Box;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JScrollPane;
+import javax.swing.JToolBar;
+import org.jdesktop.swingx.JXTable;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.ExtensionHookView;
+import org.parosproxy.paros.extension.SessionChangedListener;
+import org.parosproxy.paros.extension.history.ExtensionHistory;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.model.HistoryReferenceEventPublisher;
+import org.parosproxy.paros.model.Session;
+import org.zaproxy.zap.ZAP;
+import org.zaproxy.zap.utils.TableExportButton;
+
+public class ExtensionAllInOneNotes extends ExtensionAdaptor implements SessionChangedListener {
+
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionAllInOneNotes";
+    protected static final String PREFIX = "allinonenotes";
+
+    /**
+     * Relative path (from add-on package) to load add-on resources.
+     *
+     * @see Class#getResource(String)
+     */
+    private static final String RESOURCES = "resources";
+
+    private static final ImageIcon ICON =
+            new ImageIcon(ExtensionAllInOneNotes.class.getResource(RESOURCES + "/notepad.png"));
+
+    private AbstractPanel statusPanel;
+    private NotesTableModel notesTableModel = null;
+    private JXTable notesTable = null;
+    private EventConsumerImpl eventConsumerImpl;
+    private ExtensionHookView hookView;
+    private ExtensionHistory extHistory;
+    private JToolBar toolBar;
+    private JButton reload;
+    private ActionListener reloadActionListener;
+    private TableExportButton<JXTable> exportButton = null;
+
+    public ExtensionAllInOneNotes() {
+        super(NAME);
+        setI18nPrefix(PREFIX);
+    }
+
+    @Override
+    public void init() {
+        super.init();
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+        // As long as we're not running as a daemon
+        if (getView() != null) {
+            extensionHook.addSessionListener(this);
+            hookView = extensionHook.getHookView();
+            hookView.addStatusPanel(getStatusPanel());
+            eventConsumerImpl = new EventConsumerImpl(getNotesTableModel());
```

Comment:
This needs to be done before the previous statement (the call to `getStatusPanel()` ends up using `eventConsumerImpl`).